### PR TITLE
Update Web Components for Agent pages

### PR DESCRIPTION
**Pull request by tunzor**
This PR is specifically for the [`agent` pages](https://www.consul.io/docs/agent) of the docs.
- Added `<CodeBlockConfig>` tags for code snippets with headings where applicable
- Added missing `$` to applicable shell-session examples

### CODE REVIEW

File: website/content/docs/agent/index.mdx [hunk 1]
```diff
@@ -167,7 +167,7 @@ This is because clients are most commonly used to register services in the Consu
 The following example starts a Consul agent that takes configuration settings from a file called `server.json` located in the current working directory:
 
 ```shell-session
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/index.mdx:169]:
```suggestion
```shell-session hideClipboard
```

File: website/content/docs/agent/options.mdx [hunk 2]
```diff
@@ -66,10 +66,9 @@ The options below are all specified on the command-line.
   state as other nodes will treat the non-routability as a failure. In Consul 1.1.0 and later this can be dynamically defined with a [go-sockaddr]
   template that is resolved at runtime.
 
-  <CodeBlockConfig>
+  <CodeBlockConfig heading="Using a static network interface name">
 
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:71]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 3]
```diff
@@ -113,21 +112,30 @@ The options below are all specified on the command-line.
   both. If you have any firewalls, be sure to allow both protocols. In Consul 1.1.0 and later this can be dynamically defined with a [go-sockaddr]
   template that must resolve at runtime to a single address. Some example templates:
 
+  <CodeBlockConfig heading="Using address within a specific CIDR">
+  
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:117]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 4]
```diff
@@ -113,21 +112,30 @@ The options below are all specified on the command-line.
...
-  # Using address within a specific CIDR
   $ consul agent -bind '{{ GetPrivateInterfaces | include "network" "10.0.0.0/8" | attr "address" }}'
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using a static network interface name">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:125]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 5]
```diff
@@ -113,21 +112,30 @@ The options below are all specified on the command-line.
...
-  # Using a static network interface name
   $ consul agent -bind '{{ GetInterfaceIP "eth0" }}'
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using regular expression matching for network interface name that is forwardable and up">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:133]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 6]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
 
   Here are some examples of using `-retry-join`:
 
+  <CodeBlockConfig heading="Using a DNS entry">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:375]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 7]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
...
-  # Using a DNS entry
   $ consul agent -retry-join "consul.domain.internal"
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using IPv4">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:383]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 8]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
...
-  # Using IPv4
   $ consul agent -retry-join "10.0.4.67"
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using a non-default Serf LAN port">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:391]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 9]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
...
-  # Using a non-default Serf LAN port
   $ consul agent -retry-join "192.0.2.10:8304"
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using IPv6">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:399]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 10]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
...
-  # Using IPv6
   $ consul agent -retry-join "[::1]:8301"
   ```
 
+  </CodeBlockConfig>
+
+  <CodeBlockConfig heading="Using multiple addresses">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:407]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/options.mdx [hunk 11]
```diff
@@ -362,43 +370,61 @@ The options below are all specified on the command-line.
...
   ### Cloud Auto-Joining
 
   As of Consul 0.9.1, `retry-join` accepts a unified interface using the
   [go-discover](https://github.com/hashicorp/go-discover) library for doing
   automatic cluster joining using cloud metadata. For more information, see
   the [Cloud Auto-join page](/docs/agent/cloud-auto-join).
 
+  <CodeBlockConfig heading="Using Cloud Auto-Joining">
+
   ```shell
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/options.mdx:422]:
```suggestion
  ```shell-session
```

File: website/content/docs/agent/sentinel.mdx [hunk 12]
```diff
@@ -58,9 +62,9 @@ The following are two examples of ACL policies with Sentinel rules.
 
 ### Required Key Suffix
 
-Any values stored under the key prefix "dc1" must end with "dev"
+<CodeBlockConfig heading="Any values stored under the key 'dc1' end with 'dev'">
 
-```text
+```hcl
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/sentinel.mdx:67]:
```suggestion
```sentinel
```

Sentinel might not be value so text might be a better option.

File: website/content/docs/agent/sentinel.mdx [hunk 13]
```diff
@@ -72,11 +76,13 @@ EOF
...
 
+</CodeBlockConfig>
+
 ### Restricted Update Time
 
-The key "haproxy_version" can only be updated during business hours.
+<CodeBlockConfig heading="The key 'haproxy_version' can only be updated during business hours">
 
-```text
+```hcl
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/sentinel.mdx:85]:
```suggestion
```sentinel
```

**Review by karl-cardenas-coding — APPROVED** [review]
Minor suggestions related to the components

File: website/content/docs/agent/sentinel.mdx [hunk 14]
```diff
@@ -21,7 +21,9 @@ Sentinel policies are applied during writes to the KV Store.
 
 An optional `sentinel` field specifying code and enforcement level can be added to [ACL policy definitions](/docs/agent/acl-rules#sentinel-integration) for Consul KV. The following policy ensures that the value written during a KV update must end with "dc1".
 
-```text
+<CodeBlockConfig heading="Ensure values written during KV updates end in 'dc1'">
+
+```hcl
```
Comment (tunzor) [inline website/content/docs/agent/sentinel.mdx:26]:
```suggestion
```sentinel
```

File: website/content/docs/agent/sentinel.mdx [hunk 15]
```diff
@@ -21,7 +21,9 @@ Sentinel policies are applied during writes to the KV Store.
 
 An optional `sentinel` field specifying code and enforcement level can be added to [ACL policy definitions](/docs/agent/acl-rules#sentinel-integration) for Consul KV. The following policy ensures that the value written during a KV update must end with "dc1".
 
-```text
+<CodeBlockConfig heading="Ensure values written during KV updates end in 'dc1'">
+
+```sentinel
```
Comment (tunzor) [inline website/content/docs/agent/sentinel.mdx:26]:
```suggestion
```go
```

File: website/content/docs/agent/sentinel.mdx [hunk 16]
```diff
@@ -58,9 +62,9 @@ The following are two examples of ACL policies with Sentinel rules.
 
 ### Required Key Suffix
 
-Any values stored under the key prefix "dc1" must end with "dev"
+<CodeBlockConfig heading="Any values stored under the key 'dc1' end with 'dev'">
 
-```text
+```sentinel
```
Comment (tunzor) [inline website/content/docs/agent/sentinel.mdx:67]:
```suggestion
```go
```

File: website/content/docs/agent/sentinel.mdx [hunk 17]
```diff
@@ -72,11 +76,13 @@ EOF
...
 
+</CodeBlockConfig>
+
 ### Restricted Update Time
 
-The key "haproxy_version" can only be updated during business hours.
+<CodeBlockConfig heading="The key 'haproxy_version' can only be updated during business hours">
 
-```text
+```sentinel
```
Comment (tunzor) [inline website/content/docs/agent/sentinel.mdx:85]:
```suggestion
```go
```

**Review by karl-cardenas-coding — APPROVED** [review]
👍🏼

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/553336.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 84234e7800f61139efd67abda84ac040e67ae15c onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 84234e7800f61139efd67abda84ac040e67ae15c onto `release/1.11.x` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 84234e7800f61139efd67abda84ac040e67ae15c onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/553336)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 84234e7800f61139efd67abda84ac040e67ae15c onto `release/1.9.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/553336)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 84234e7800f61139efd67abda84ac040e67ae15c onto `release/1.8.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/553336)

**Pull request merged**